exercises/flatten-array/FlattenArrayExample.swift

```diff
@@ -0,0 +1,42 @@
+import Foundation
+
+func constructList(elements: Any?...) -> [Any?] {
+
+    return elements
+}
+
+func flattenArray<T>(_ list: [Any?]) -> [T] {
+
+    var flattenedArray = [T]()
+
+    func extractArrayElements(array: [Any?]) {
+
+        for element in array {
+
+            if let j = element {
+
+                let anyObjectArray = j as? [Any?]
+
+                if let unwrappedArray = anyObjectArray {
+
+                    extractArrayElements(array: unwrappedArray)
```

Comment:
I think we can remove the nested `extractArrayElements` function here and just recursively call the flatten array function. 

I see the reason you wanted/needed to do it this way is you are "closing over" the `flattenedArray` variable. ... both solutions are valuable.